src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/LimitedUseItem.java

```diff
@@ -0,0 +1,156 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.apache.commons.lang.Validate;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Sound;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.persistence.PersistentDataContainer;
+import org.bukkit.persistence.PersistentDataType;
+
+import io.github.thebusybiscuit.cscorelib2.chat.ChatColors;
+import io.github.thebusybiscuit.slimefun4.api.SlimefunAddon;
+import io.github.thebusybiscuit.slimefun4.api.items.ItemState;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.implementation.items.magical.staves.StormStaff;
+import io.github.thebusybiscuit.slimefun4.utils.LoreBuilder;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+
+/**
+ * This class represents an item with a limited number of uses.
+ * When the item runs out of "uses", it breaks.
+ *
+ * @author Linox
+ * @author Walshy
+ * @author TheBusyBiscuit
+ * @author martinbrom
+ *
+ * @see StormStaff
+ */
+public abstract class LimitedUseItem extends SimpleSlimefunItem<ItemUseHandler> {
```

Comment:
Why is this abstract...?

```suggestion
public class LimitedUseItem extends SimpleSlimefunItem<ItemUseHandler> {
```

Reply:
It is abstract because it does not implement the `ItemUseHandler`, which is something each limited item must define itself, I feel like there is no good default behavior.

Reply:
I see :+1: